### PR TITLE
feat: added authentication

### PR DESCRIPTION
**Pull request by varnita619**
added authentication in
- signup page
- login page
- logout page

Screenshot (20)

### CODE REVIEW

File: src/Context/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const AuthContext = createContext();
+
+const AuthContextProvider = ({ children }) => {
+    const getToken = localStorage.getItem("login-Token");
+    const getUserDetails = localStorage.getItem("user");
+
+    const [token, setToken] = useState(getToken || "");
+    const [user, setUser] = useState(getUserDetails || "");
+
```
Comment (anuj-kumary) [inline src/Context/AuthContext.js:12]: Instead of giving OR you can use optional chaining

File: src/Pages/Signup/Signup.jsx [hunk 2]
```diff
@@ -1,57 +1,153 @@
...
+                Terms & Privacy
+              </a>
+              .
+            </p>
+
+            <div className="clearfix">
+              <button
+                type="submit"
+                className="signupbtn anchor-btn"
+                // onClick={(e) => signUpFormHandler(e)}
```
Comment (anuj-kumary) [inline src/Pages/Signup/Signup.jsx]: Remove unwanted comments

**Review by anuj-kumary — APPROVED** [review]

File: src/Pages/Signup/Signup.jsx [hunk 3]
```diff
@@ -1,57 +1,153 @@
...
+  };
+
+  return (
+    <>
+      <form onSubmit={(e) => signUpFormHandler(e)}>
+        <div className="auth-container">
+          <div className="signup-container">
+            <h1>Sign Up</h1>
+            <p>Please fill in this form to create an account.</p>
+            {/* <hr /> */}
```
Comment (Rohanmond) [inline src/Pages/Signup/Signup.jsx]: remove unwanted comments

**Review by Rohanmond — APPROVED** [review]

File: src/Context/AuthContext.js (hunk 1 above)
Comment (varnita619) [reply src/Context/AuthContext.js:12]: ok sure

**Pull request merged**